ntbs-service/DataMigration/MigrationRepository.cs

```diff
@@ -53,6 +54,13 @@ public class MigrationRepository : IMigrationRepository
             WHERE notificationInRange.NotificationDate >= @StartDate AND notificationInRange.NotificationDate < @EndDate
             AND NOT EXISTS ({_importHelper.SelectImportedNotificationWhereIdEquals("notificationInGroup.OldNotificationId")})";
 
+        private string LegacyNotificationByNhsNumberQuery => $@"
+            SELECT n.OldNotificationId, n.Source
+            FROM MigrationNotificationsView n
+            LEFT JOIN Demographics dmg ON dmg.OldNotificationId = n.OldNotificationId
+            WHERE dmg.NhsNumber = @NhsNumber
+            AND NOT EXISTS ({_importHelper.SelectImportedNotificationWhereIdEquals("n.OldNotificationId")})";
```

Comment:
Have you tested the performance of this query in the live database? How long does it take?

Performance is more important than usual for this endpoint, as we want the page to be responsive (i.e. we don't want a user to enter an NHS number and then move onto later fields before seeing the warning)

Reply:
Also: don't we need to exclude deleted and denotified records here?

Reply:
Yep, excluding denotified records and actually taking the JOIN out, as we have NhsNumber in the view, brings the runtime on the query to between 1 and 2 seconds.

Reply:
Hmm... that's a lot slower than what we should be aiming for :-(

I've had a look into this and it looks like going via the `MigrationNotificationsView` is just too slow. Fortunately, I think all the information we need is in the `Demographics` table anyway, so I think we can just get it from there.

It seems to be about 30ms doing it that way.

Reply:
What were you joining on Demographics to get the notifiation's denotified state?

Reply:
I don't know what Josh used, but you could use `MergedNotifications` if you are worried? `MergedNotifications` is a table, and `MigrationNotificationsView ` is `MergedNotifications` joined onto all of the various details tables, so by joining the two you essentially get a stripped down version of `MigrationNotificationsView` just containing the stuff you see.

Another performance improvement I can spot is using `INNER` rather than `LEFT` joins. These are invariably *much* quicker, and if there was a notification without demographics (which I'm fairly sure is impossible) then we wouldn't care about it anyway because we are just searching for NHS numbers.

Reply:
Ah yes, I had forgotten about the notification status as I was testing, but as Benji already pointed out, it isn't much slower when joining with `MergedNotifications`